server/controller/userController.js

```diff
@@ -90,34 +91,29 @@ class Users {
    * @param {function} next
    */
   static async resetPassword(req, res, next) {
-    const { email } = req.body;
+    const { email } = req.params;
     const foundUser = await UserModel.find(email);
     if (!foundUser) { return errorRes(next, 404, 'User with this email was not found'); }
-    const token = jwt.sign({ email }, process.env.SECRETkey, { expiresIn: '3h' });
-    const sent = await resetPasswordEmail(email, token);
-    if (!sent) return errorRes(next, 500, 'Unable to send Email');
-    return successRes(res, 200, { message: 'Reset Password Email Successfully Sent' });
-  }
-
-  /**
-   * Change a users password
-   * @param {object} req - request
-   * @param {object} res - response
-   * @param {function} next
-   */
-  static async changePassword(req, res, next) {
-    const { password } = req.body;
-    const { token } = req.query;
-    try {
-      const { email } = jwt.verify(token, process.env.SECRETkey);
-      const foundUser = await UserModel.find(email);
-      if (!foundUser) return errorRes(next, 404, 'User with this email was not found');
+    if (Object.keys(req.body).length === 0) {
+      const password = generator.generate({
+        length: 10,
+        strict: true,
+      });
       const hashedPassword = bcrypt.hashSync(password, 8);
       const update = await UserModel.changePassword(email, hashedPassword);
-      return successRes(res, 200, update);
-    } catch (error) {
-      next(error);
+      const sent = await resetPasswordEmail(email, password);
```

Comment:
Is there something wrong with this logic?

You updated the user's password in the database and then tries to send an email. What happens when the user doesn't get the `auto-generated password` via their email?

Reply:
I can see the flaw in the logic. From the ADC we are to send back a password. What if I send the users current password back to the user's email, instead of generating and changing the user's password.

Reply:
How will you get the user's current password considering it has been hashed? Are you storing passwords as plain text in your db?

Reply:
that's true, I can't unhash the passwords that have been hashed.
On a second taught, I'll change the logic to send the email first. if the email is successfully sent, then the generated password should be stored in the database. If for any reason the user doesn't get the email, the user would just ask for another password reset, which would repeat the process.
Is that a good way to go about it?

Reply:
Maybe that's a good approach.

An alternate solution is to augment your current logic,

1. Retrieve the users' current (hashed) password from the db, store in a variable
2. Generate a new password and update the db
3. Send an email with the newly generated password
4. If mail fails to send, update the newly generated password in db with old password retrieved in `step 1`

Reply:
This way, you restore the db to its original state

Reply:
Ok. I'll implement it that way. Thank you

Reply:
I have made the changes